tools/docs/example_test_helper/src/lib.rs

```diff
@@ -326,6 +327,12 @@ impl CmdRunner {
     }
 }
 
+pub fn find_available_port() -> Result<u16> {
+    let listener = TcpListener::bind("127.0.0.1:0")?;
+    let address = listener.local_addr()?;
+    Ok(address.port())
+}
```

Comment:
An optional improvement: since we are using this function exclusively in tests, you can move the `.expect()` calls here and return a `u16`, so that, in the tests, you can just call the function and not duplicate the `.expect("Failed to find available port");` piece.

Reply:
Done